### PR TITLE
Add Sonatype publications to CI

### PR DESCRIPTION
**Pull request by harlem88**
Add a github action to automatize the publication of the Sonatype nexus staging repository. The action will trigger when a release or draft of a release is published, or a pre-release is changed to a release.

Note: a manual interaction is still required to release the package from the staging repository to the public one. This step is deliberately left to an admin user because is not possible to remove a packet published by mistake.

### CODE REVIEW

File: scripts/publish-root.gradle [hunk 1]
```diff
@@ -14,12 +13,11 @@ if (secretPropsFile.exists()) {
...
-    ext["ossrhPassword"] = System.getenv('OSSRH_PASSWORD')
-    ext["sonatypeStagingProfileId"] = System.getenv('SONATYPE_STAGING_PROFILE_ID')
-    ext["signing.keyId"] = System.getenv('SIGNING_KEY_ID')
-    ext["signing.password"] = System.getenv('SIGNING_PASSWORD')
-    ext["signing.secretKeyRingFile"] = System.getenv('SIGNING_SECRET_KEY_RING_FILE')
+    ext["ossrhUsername"] = System.getenv("OSSRH_USERNAME")
+    ext["ossrhPassword"] = System.getenv("OSSRH_PASSWORD")
+    ext["sonatypeStagingProfileId"] = System.getenv("SONATYPE_STAGING_PROFILE_ID")
+    ext["signing.key"] = System.getenv("SIGNING_KEY")
+    ext["signing.password"] = System.getenv("SIGNING_PASSWORD")
```
Comment (lucaato) [inline scripts/publish-root.gradle:20]: to me it looks like this env variable is not set in the action, is this correct ?

Comment (harlem88) [reply scripts/publish-root.gradle:20]: Is it set here https://github.com/astarte-platform/astarte-device-sdk-java/blob/cc2677e439108d2a2a5842c241ec27e924ea185f/.github/workflows/sonatype-publish.yaml#L42?

Comment (lucaato) [reply scripts/publish-root.gradle:20]: You are setting `OSSRH_PASSWORD` there, but the code looks for `SIGNING_PASSWORD` or am I missing something ?

Comment (harlem88) [reply scripts/publish-root.gradle:20]: Ah ok i will add also `SIGNING_PASSWORD` to the system env

**Review by lucaato — APPROVED** [review]

**Pull request merged**